crypto-core/src/main/java/com/palantir/crypto/cipher/CipherStreamSupplier.java

```diff
@@ -30,12 +35,14 @@
     /**
      * Wraps the {@link InputStream} in a {@link CipherInputStream} which uses the given {@link Cipher} for decryption.
      */
-    CipherInputStream getInputStream(InputStream is, Cipher cipher);
+    CryptoInputStream getInputStream(String transformation, Properties props, InputStream is,
```

Comment:
what does `transformation` mean, for instance? would prefer to define this interface as narrowly as possible and hide away the somewhat ugly API for the underlying apache commons class.